### PR TITLE
Update database for 2.4 leaks

### PR DESCRIPTION
**Pull request by bschouwerwou**
- Adds upcoming 2.4 characters with leaked details, Shenhe and Yun Jin
- Adds leaked 2.4 weapon, Calamity Queller

Note that this is branched off #104 since it's utilizing the leaked weapon flag that introduces.

### CODE REVIEW

File: web/db/characterMaterials.ts
Comment (luaneko) [inline web/db/characterMaterials.ts:163]: I noticed that Dragonheir's False Fin for Shenhe was not added in this PR, so I added it myself. My assumption is that the drop boss details are not available yet, so you chose to skip it for now. Is that correct?

Personally I would prefer the character profile to be complete, even if the materials aren't acquirable right now. With my changes, under Shenhe's profile, the item is now displayed like this:

image

What do you think? Is this better than not showing the item at all, or would it instead cause more confusion for the users? I think it should be fine since there's a strong accuracy disclaimer at the top of the page.

File: web/db/weapons.ts
Comment (luaneko) [inline web/db/weapons.ts:148]: This item leads to [this wiki page](https://genshin-impact.fandom.com/wiki/Calamity_Queller) which doesn't exist yet. It should be solved when the wiki itself is updated for this weapon after some time. Just leaving this here as a remark, since it's not actionable.

File: web/db/characterMaterials.ts
Comment (bschouwerwou) [reply web/db/characterMaterials.ts:163]: > My assumption is that the drop boss details are not available yet, so you chose to skip it for now. Is that correct?

That is correct, and I wasn't sure if leaving it out or not was preferable. Even if the boss details were available, I'd still want to hold off on including that part so we don't give the impression that the item is acquirable when it isn't.

I agree it should be fine to include just the item though given the disclaimer. 

We could consider adjusting that message to make it slightly more clear though too, since I can't think of another case when an item would have this message shown. Are there any other cases? Even a small update like `This item does not currently drop from any domains.` helps suggest that it will change eventually and reduce the chance of confusion.

Comment (luaneko) [reply web/db/characterMaterials.ts:163]: > This item does not currently drop from any domains.

Sounds good!

File: web/components/ChangelogModal.tsx
Comment (bschouwerwou) [inline web/components/ChangelogModal.tsx]: So I made an assumption about the date for this, but it's now incorrect.

@phosphene47, should I leave adding items to the changelog to you so it can be more correct based on when you're going to deploy?

Comment (luaneko) [reply web/components/ChangelogModal.tsx]: Sure. The changelog doesn't have to be super accurate though; it can be off by a day or two. The "date" doesn't have any specific definition right now. It could be the update deployment date or PR submission date in some arbitrary timezone.

As long as the change is listed here, it's fine. If it isn't, I'll update it myself just before merging.

**Review by luaneko — APPROVED** [review]
LGTM

**Pull request merged**
